dev-learning/Controllers/UsersController.cs

```diff
@@ -158,14 +188,35 @@ public async Task<ActionResult<User>> GetCurrentUser()
                 return Unauthorized();
             }
         }
-
+        
         // GET: api/Users/team_users/5
         [HttpGet("team_users/{teamId}")]
         public async Task<List<User>> GetTeamUsers(int teamId)
         {
             return await _context.Users.Where(c => c.TeamId == teamId).ToListAsync();
         }
 
+        private async Task<List<CalendarDay>> GetUserCalendar(int id, int days)
+        {
+            var calendar = new List<CalendarDay>();
+
+            var userSubjects = await _context.UserSubjects.Include(x => x.Subject).Where(u => u.UserId == id).ToListAsync();
+
+            for (int i = 1; i <= days; i++)
+            {
+                var calendarDay = new CalendarDay(i);
+                var currentDate = new DateTime(DateTime.Now.Year, DateTime.Now.Month, i);
+                var currentUserSubjects = userSubjects.Where(x => x.StartDateTime <= currentDate).Where(y => y.EndDateTime >= currentDate).ToList();
+
+                for(int x = 0; x < currentUserSubjects.Count; x++)
+                {
+                    calendarDay.AddSubject(currentUserSubjects[x].Subject, currentUserSubjects[x].IsLearned);
```

Comment:
I need subject id as well

Reply:
If a user clicks on the subject I want to open subject by its id. That's just one example there are more use-cases where id is required